adr/adr.001.folder structure/adr.001.folder structure.md

```diff
@@ -0,0 +1,260 @@
+
+Each feature directory may contain any of the following directories:
+
+- `components`: Small, feature-specific UI elements. Always check
+  the shared design system before creating new components.
+  Consult with Andrew and Thierry if unsure about existing
+  components.
+
+- `screens`: Top-level components that encapsulate entire
+  device screens.
+
+- `navigation`: Contains setup for navigation system, including
+  deep linking and navigation parameters.
+
+- `hooks`: React hooks specific to the feature. Use caution
+  when adding hooks, as we're transitioning to state machines
+  for business logic.
+
+- `utils`: One-off utility functions relevant to the feature.
+
+- `data`: Contains Zustand queries and data mapping methods.
+
+- `clients`: Encapsulate feature behavior and control
+  dependencies used to implement the feature.
+
+- `queries`: React Query functionality for caching, refetching,
+  and optimistic updates specific to the feature.
+
+- `types`: TypeScript types and interfaces relevant to the feature.
+
+- `actors`: State machines or actors that control feature behavior.
+
+
+## Considered Options
+
+1. **Keep the current structure (organized by file type)**
+2. **Reorganize code by features (package by feature)**
+3. **Gradual approach (combination of file type and feature-based organization, working towards a package by feature structure)**
+
+## Pros and Cons of the Options
+
+### Option 1: Keep the current structure
+
+*Good, because:*
+
+- Familiar to current developers
+- No immediate refactoring needed
+
+*Bad, because:*
+
+- Scattered codebase makes it hard to find related files
+- Reduces productivity with AI-assisted tools
+- Difficult to maintain as the codebase grows
+
+### Option 2: Reorganize code by features
+
+*Good, because:*
+
+- Groups related code together, improving maintainability
+- Enhances productivity with AI-assisted tooling
+- Easier onboarding for new developers
+
+*Bad, because:*
+
+- Significant upfront refactoring effort
+- Potential for merge conflicts during transition
+- Requires updating import paths throughout the codebase
+
+### Option 3: Gradual approach
+
+*Good, because:*
+
+- Allows gradual transition
+- Balances familiarity with improved organization
+- Minimizes disruption to ongoing development
+
+*Bad, because:*
+
+- Can lead to inconsistent structure during transition
+- Might still scatter related code temporarily
+- Requires ongoing attention to migration
+
+## Proposed Option: Option 3 - Gradual approach
+
+We chose to reorganize the codebase by features gradually
+because it offers the benefits of improved organization
+while being more feasible to implement. A complete
+refactoring at once is impractical due to the size of our
+codebase and ongoing development needs. This approach allows
+us to incrementally improve our codebase structure as we
+touch different parts of the code.
+
+## Implementation Details
+
+- **Step 1:** Create a `features/` directory at the root of
```

Comment:
Should we accept this proposal, I'll create the skeleton directories, into which we can place files moving forward based on this structure.